### ISSUE TITLE
[Performance] Timm_models beit_base, gmixer_2's performance dropped >20% compared to v2.1

### ISSUE THREAD

**Issue by ESI-SYD** [intel/intel-xpu-backend-for-triton#708]
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 beit_base_patch16_224

<img width="1180" alt="image" src="https://github.com/intel/intel-xpu-backend-for-triton/assets/76785525/c28e43a9-274a-4294-a9d1-aa2c75b2b766">

**Comment by chengjunlu** [comment]
Quick check the E2E performance profile status.

The eager mode has almost the same kernel performance when the kineto profiler is enabled:

```
LLVM-TARGET:
Self CPU time total: 731.982ms
Self XPU time total: 149.012ms
SPIRV-TARGET:
Self CPU time total: 687.606ms
Self XPU time total: 148.457ms
``` 

But the CPU host time is unreasonable when the kineto profiler is enabled. We can skip the host time for now.

The graph mode has almost the same kernel performance too.
```
LLVM-TARGET:
Self CPU time total: 681.791ms
Self XPU time total: 126.993ms
gemm_kernel         0.00%       0.000us         0.00%       0.000us       0.000us      69.137ms        54.44%      69.137ms     315.694us           219
Triton kernel total time is: 58ms

SPIRV-TARGET:
Self CPU time total: 544.637ms
Self XPU time total: 124.709ms
gemm_kernel         0.00%       0.000us         0.00%       0.000us       0.000us      68.894ms        55.24%      68.894ms     314.584us           219  
Triton kernel total time is: 55ms
```

Need to check the GPU bubble time further.

**Comment by chengjunlu** [comment]
The performance gap is caused by the same issue as #707.
The performance of the LLVM target is good as SPIRV target with the torch/ipex.

Torch branch: https://github.com/intel-innersource/frameworks.ai.pytorch.private-gpu/tree/dev/triton-test-3.0
IPEX branch: https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-gpu/tree/dev/triton-test-3.0
IGC driver version: `ii  libigc1                              1.0.15136.24-775~22.04                  amd64        Intel graphics compiler for OpenCL -- core libs`

The performance of the 
```
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 beit_base_patch16_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 gmixer_24_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 mixer_b16_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 swin_base_patch4_window7_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 gmlp_s16_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 sebotnet33ts_256
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 tnt_s_patch16_224
bash inductor_xpu_test.sh timm_models amp_bf16 training performance xpu 0 static 1 0 ese_vovnet19b_dw
```

```
beit_base_patch16_224: 1.154x
gmixer_24_224: 1.906x
mixer_b16_224: 1.224x
```

**Comment by chengjunlu** [comment]
There are two issues found:
1.	One IPEX commit set the default log level to WARN accidently. Unexpected L0 barrier causes the GPU bubble time increased.
2.	The IPEX bench utils uses the SYCL barrier to measure the Triton kernel time. But it has issue in successfully to measure the Triton kernel. That causes autotune may chose a sub-optimal Triton kernel to be used.

The first issue has already been fixed on IPEX dev/triton-test-3.0 branch.
The second issue is the oneAPI 2024.0 regression issue. It can be resolved by setting `UR_L0_IN_ORDER_BARRIER_BY_SIGNAL=0`.
